frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -161,6 +166,42 @@ private void subscribeToGitRepositoryFilesChanges() {
     Disposer.register(project, messageBusConnection);
   }
 
+  private void subscribeToVcsBranchChange() {
+    Topic<BranchChangeListener> topic = BranchChangeListener.VCS_BRANCH_CHANGED;
+    BranchChangeListener listener = new BranchChangeListener() {
+      @Override
+      public void branchWillChange(String branchName) {}
+
+      @Override
+      public void branchHasChanged(String branchName) {
+        if (gitMacheteRepositorySnapshot != null) {
+          IBranchLayoutEntry entry = gitMacheteRepositorySnapshot.getBranchLayout().findEntryByName(branchName);
+          if (entry == null) {
+            notifyAboutNonManagedBranch(branchName);
+          }
+        }
+      }
+
+      private void notifyAboutNonManagedBranch(String branchName) {
+
+        Notification notification = NOTIFICATION.createNotification(
+            format("action.GitMachete.EnhancedGraphTable.non-managed-branch-notification.text", branchName),
+            NotificationType.INFORMATION);
+        notification.addAction(
+            new NotificationAction(getString("action.GitMachete.EnhancedGraphTable.non-managed-branch-notification.action")) {
+              @Override
+              public void actionPerformed(AnActionEvent e, Notification notification) {
+                System.out.println("sliding in...");
+              }
+            });
+        notification.notify(project);
```

Comment:
I wonder if `notification.notify` could be placed in `actionPerformed`
Would it be triggered from here?